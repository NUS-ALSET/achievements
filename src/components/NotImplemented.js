/**
 * @file NotImplemented container module
 * @author Theodor Shaytanov <theodor.shaytanov@gmail.com>
 * @created 07.03.18
 */

import React from "react";
import PropTypes from "prop-types";

class NotImplemented extends React.PureComponent {
  static propTypes = { stub: PropTypes.bool };

  render() {
    return <div>Not Implemented</div>;
  }
}

export default NotImplemented;
