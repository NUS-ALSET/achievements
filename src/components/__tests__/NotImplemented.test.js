import React from "react";
import { createShallow } from "@material-ui/core/test-utils";
import NotImplemented from "../NotImplemented";

describe("<NotImplemented />", () => {

  let shallow;
  beforeEach(() => {
    shallow = createShallow();
  });

  it("Should test NotImplemented component", () => {
    const component = shallow(
      <NotImplemented />
    );
    expect(component).toMatchSnapshot();
  });

});
