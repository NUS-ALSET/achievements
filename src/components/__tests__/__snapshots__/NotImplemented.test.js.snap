// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotImplemented /> Should test NotImplemented component 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NotImplemented />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "Not Implemented",
    },
    "ref": null,
    "rendered": "Not Implemented",
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Not Implemented",
      },
      "ref": null,
      "rendered": "Not Implemented",
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "context": Object {},
  },
}
`;
